File: core4/leetcode/Two_Pointers/N4/N4.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.size();
        int m = word2.size();
        
        string ans = "";
        int i = 0;
        
        while(i < n && i < m){
            ans += word1[i] ;
            ans += word2[i];
            i++;
        }
        
        while(i < n){
            ans += word1[i];
            i++;
        }
        
        while(i < m){
            ans += word2[i];
            i++;
        }
        
        return ans;
    }
};

File: core4/leetcode/Two_Pointers/N4/N4.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        n = len(word1)
        m = len(word2)
        
        i = 0 
        ans = ""
        
        while i < n and i < m :
            ans += word1[i]
            ans += word2[i]
            i+=1 
            
        while i < n:
            ans += word1[i]
            i+=1 
            
        while i < m:
            ans += word2[i]
            i+=1 
            
        return ans 
            
        
